Fixed FortuneAbility.

ArtiOJ opened the pull request:
Made FortuneAbility work as intended.

ArtiOJ committed:
Fixed FortuneAbility.

RTClarkV commented on src/main/java/dev/corestone/lotrings/abilities/FortuneAbility.java:
```diff
@@ -36,17 +37,25 @@ public void switchState(RingState ringState) {
+        event.getBlock().breakNaturally(tool);
     }
-}
+
```
does this not damage the player's tool?

RTClarkV commented:
does this not damage the player's tool?

ArtiOJ committed:
Fixed FortuneAbility (again).

ArtiOJ commented:
Fixed @RTClarkV

ArtiOJ committed:
Added lima's changes, with some modifications.

ArtiOJ commented:
I just added all lima's changes to this, because I am bad and do not know how to push to another branch.

RTClarkV commented on src/main/java/dev/corestone/lotrings/abilities/NecromanceAbility.java:
```diff
@@ -0,0 +1,106 @@
+    private String summonType;
+    private double summonHP;
+    private int summonLength;
+    private HashMap<Player, List<LivingEntity>> necroSummons = new HashMap<>();
```
the hash map here is completely unnecessary. This ability and class is tied to the player's ring. You can literally do ring.getOwner() to get the UUID of the ring's owner. All you need is a list of the entities. If another ring also has this ability, it makes a new NecromancyAbility.java class for that player.

ArtiOJ replied:
Did not catch that in my code review, I will fix this when I get home.

RTClarkV commented on src/main/java/dev/corestone/lotrings/abilities/NecromanceAbility.java:
```diff
@@ -0,0 +1,106 @@
+    public void onHit(EntityDamageByEntityEvent event){
+        if (event.getDamager() instanceof Player && event.getEntity() instanceof LivingEntity){
+            Player player = ((Player) event.getDamager());
+            System.out.println("Damager: "+player);
```
I think some debugging code was left here.

ArtiOJ committed:
Fixed NecromanceAbility

ArtiOJ commented:
Think this works, read it over.

ArtiOJ committed:
Added customizable damage to LightningAbility

ArtiOJ committed:
Hotfix for FireballRain

Theoretically stopped the player from taking at least the non-explosion or fire damage from their own fireballs. (UNTESTED)
